ui/src/adapters/parsers/forms.ts

```diff
@@ -69,10 +70,10 @@ interface LinkExpiration {
 }
 
```

Comment:
I like the name changes and organization you're proposing here.

To be consistent with the `1 type <-> 1 parser` non-written rule and to simplify the reading of the code, I'd like to have a separate parser for `LinkExpiration` that we use later with an intersection as we do with the types.

```typescript
interface LinkExpiration {
  linkExpirationDate?: dayjs.Dayjs | null;
  linkExpirationTime?: dayjs.Dayjs | null;
}

const linkExpirationParser = getStrictParser<LinkExpiration>()(
  z.object({
    linkExpirationDate: dayjsInstanceParser.nullable().optional(),
    linkExpirationTime: dayjsInstanceParser.nullable().optional(),
  })
);

export type IssuanceMethodFormData =
  | (LinkExpiration & {
      linkMaximumIssuance?: number;
      type: "credentialLink";
    })
  | {
      did: string;
      type: "directIssue";
    };

const issuanceMethodFormDataParser = getStrictParser<IssuanceMethodFormData>()(
  z.union([
    linkExpirationParser.and(
      z.object({
        linkMaximumIssuance: z.number().optional(),
        type: z.literal("credentialLink"),
      })
    ),
    z.object({
      did: z.string(),
      type: z.literal("directIssue"),
    }),
  ])
);
```